### PR TITLE
Clean up rs exports

### PR DESCRIPTION
**Pull request by sieniven**
## Summary

- Clean up rs exports FFIs to only have exposed FFI functions should be stored inside the library.
- Refactor bridge_to_dst20 and create_dst20 functions implementation into exposed FFI functions.

## Implications

- Storage
  - [ ] Database reindex required
  - [ ] Database reindex optional
  - [ ] Database reindex not required
  - [x] None

- Consensus
  - [ ] Network upgrade required
  - [ ] Includes backward compatible changes
  - [ ] Includes consensus workarounds
  - [ ] Includes consensus refactors
  - [x] None

### CODE REVIEW

File: lib/ain-rs-exports/src/evm.rs [hunk 1]
```diff
@@ -438,75 +437,61 @@ pub fn evm_try_get_block_number_by_hash(
...
+            }));
+
+            match SERVICES
+                .evm
+                .queue_tx(queue_id, system_tx, native_hash, U256::zero())
+            {
+                Ok(_) => cross_boundary_success(result),
+                Err(e) => cross_boundary_error_return(result, e.to_string()),
+            }
+        }
```
Comment (shohamc1) [inline lib/ain-rs-exports/src/evm.rs:463]: This is more readable IMO.

```suggestion
    let address = match ain_contracts::dst20_address_from_token_id(token_id) {
        Ok(address) => address,
        Err(e) => cross_boundary_error_return(result, e.to_string()),
    };

    debug!("Deploying to address {:#?}", address);

    let system_tx = QueueTx::SystemTx(SystemTx::DeployContract(DeployContractData {
        name: String::from(name),
        symbol: String::from(symbol),
        address,
    }));

    match SERVICES
        .evm
        .queue_tx(queue_id, system_tx, native_hash, U256::zero())
    {
        Ok(_) => cross_boundary_success(result),
        Err(e) => cross_boundary_error_return(result, e.to_string()),
    }
```

File: lib/ain-rs-exports/src/evm.rs [hunk 2]
```diff
@@ -438,75 +437,61 @@ pub fn evm_try_get_block_number_by_hash(
...
+        }));
+        match SERVICES
+            .evm
+            .queue_tx(queue_id, system_tx, native_hash, U256::zero())
+        {
+            Ok(_) => cross_boundary_success(result),
+            Err(e) => cross_boundary_error_return(result, e.to_string()),
+        }
+    } else {
+        cross_boundary_error_return(result, "Invalid address")
```
Comment (shohamc1) [inline lib/ain-rs-exports/src/evm.rs:495]: Similar thing here.

```suggestion
    let address = match address.parse() {
        Ok(address) => address,
        Err(_) => cross_boundary_error_return(result, "Invalid address"),
    };

    let contract = ain_contracts::dst20_address_from_token_id(token_id)
        .unwrap_or_else(|e| cross_boundary_error_return(result, e.to_string()));

    let system_tx = QueueTx::SystemTx(SystemTx::DST20Bridge(DST20Data {
        to: address,
        contract,
        amount: amount.into(),
        out,
    }));

    match SERVICES
        .evm
        .queue_tx(queue_id, system_tx, native_hash, U256::zero())
    {
        Ok(_) => cross_boundary_success(result),
        Err(e) => cross_boundary_error_return(result, e.to_string()),
    }
```

File: lib/ain-rs-exports/src/evm.rs [hunk 3]
```diff
@@ -438,75 +437,61 @@ pub fn evm_try_get_block_number_by_hash(
...
-        to: address,
-        contract,
-        amount: amount.into(),
-        out,
-    }));
-    SERVICES
-        .evm
-        .queue_tx(context, system_tx, native_hash, U256::zero())?;
+) {
+    if let Ok(address) = address.parse() {
```
Comment (Jouzo) [inline lib/ain-rs-exports/src/evm.rs:477]: Can use simpler `let Ok(address) = address.parse() else {  return cross_boundary_error_return(result, "Invalid address") }` here

File: lib/ain-rs-exports/src/evm.rs (hunk 2 above)
Comment (sieniven) [reply lib/ain-rs-exports/src/evm.rs:495]: Resolved with Jeremy's suggestion

File: lib/ain-rs-exports/src/evm.rs (hunk 3 above)
Comment (sieniven) [reply lib/ain-rs-exports/src/evm.rs]: Resolved in [590340a](https://github.com/DeFiCh/ain/pull/2277/commits/590340a4022596bb9f0d9dfbfa8ed49502290d01)

File: lib/ain-rs-exports/src/evm.rs (hunk 2 above)
Comment (sieniven) [reply lib/ain-rs-exports/src/evm.rs]: Resolved in [590340a](https://github.com/DeFiCh/ain/pull/2277/commits/590340a4022596bb9f0d9dfbfa8ed49502290d01)

File: lib/ain-rs-exports/src/evm.rs (hunk 1 above)
Comment (sieniven) [reply lib/ain-rs-exports/src/evm.rs]: Resolved in commit [590340a](https://github.com/DeFiCh/ain/pull/2277/commits/590340a4022596bb9f0d9dfbfa8ed49502290d01)

File: lib/ain-rs-exports/src/evm.rs [hunk 4]
```diff
@@ -438,75 +437,58 @@ pub fn evm_try_get_block_number_by_hash(
...
-    context: u64,
-    address: &str,
-    amount: [u8; 32],
-    native_tx_hash: [u8; 32],
-    token_id: &str,
-    out: bool,
-) {
-    match bridge_to_dst20(context, address, amount, native_tx_hash, token_id, out) {
-        Ok(_) => cross_boundary_success(result),
+    let address = match ain_contracts::dst20_address_from_token_id(token_id) {
```
Comment (shohamc1) [inline lib/ain-rs-exports/src/evm.rs:446]: Maybe we can use `unwrap_or_else` here similar to `evm_bridge_dst20`.

**Review by prasannavl — APPROVED** [review]

**Pull request merged**